### PR TITLE
[auth0] Add missing users_json to ImportUsersOptions

### PR DESCRIPTION
**Pull request by orangain**
Additionally, make `users` optional because either `users` or `users_json` is mandatory.

----

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://auth0.github.io/node-auth0/module-management.JobsManager.html#importUsers
  - https://github.com/auth0/node-auth0/blob/423590665c59d0832a31bfbd8474db2ef8d6ac45/src/management/JobsManager.js#L150
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Review by neshanjo — CHANGES_REQUESTED** [review]
Hi, 
thanks for you submission!

From the code in https://github.com/auth0/node-auth0/blob/423590665c59d0832a31bfbd8474db2ef8d6ac45/src/management/JobsManager.js#L177, it looks like either `users` or `users_json` must be present. With your change, this is not guaranteed anymore.

Suggestion: Create two interfaces, `ImportUsersFromFileOptions` and `ImportUsersFromJsonOptions`, one with `users`, one with `users_json`. And change 
```
    importUsers(data: ImportUsersOptions): Promise<ImportUsersJob>;
    importUsers(data: ImportUsersOptions, cb?: (err: Error, data: ImportUsersJob) => void): void;
```
to
```
    importUsers(data: ImportUsersFromFileOptions | ImportUsersFromJsonOptions): Promise<ImportUsersJob>;
    importUsers(data: ImportUsersFromFileOptions | ImportUsersFromJsonOptions , cb?: (err: Error, data: ImportUsersJob) => void): void;
```
Both interfaces may extend a common interface (e.g. `BaseImportUsersOptions`) with all shared attributes.

I'm not a TypeScript expert - so if there is a more elegant approach to support "either one or the other of two attributes", please comment!

Cheers,
Johannes

**Comment by orangain** [comment]
@neshanjo Thank you for the review! I've fixed the type definition as you suggested. I also named the union type as `ImportUsersOptions` to preserve compatibility with the previous version of the type definition.

**Comment by neshanjo** [comment]
@orangain From my side, it looks good now. Thanks again. I'm also quite new here (auth0 types commiter), so I don't know the usual process. I'd suggest that you rebase your commits into a single commit that has a message starting with `[@types/auth0]` or similar, then I'll approve the request and we'll see what happens. :)

**Comment by orangain** [comment]
@neshanjo Thank you for the suggestion. I just squashed the commit and edited its message. I misunderstood how the merge commit will be made.

From my experience, after you approve the pull request, a DefinitelyTyped maintainer will merge it and the new version of the type definition will be published automatically.

**Review by neshanjo — APPROVED** [review]

**Comment by neshanjo** [comment]
Alright, I approved the PR. Let's see what happens...

**Pull request merged**
